resources/resources.go

```diff
@@ -0,0 +1,194 @@
+}
+
+type SchedulerService interface {
+	GetSchedulerConfig() *v1beta2config.KubeSchedulerConfiguration
+	RestartScheduler(cfg *v1beta2config.KubeSchedulerConfiguration) error
+}
+
+func NewResourcesService(client clientset.Interface, pods PodService, nodes NodeService, pvs PersistentVolumeService, pvcs PersistentVolumeClaimService, storageClasss StorageClassService, schedulers SchedulerService) *Service {
+	return &Service{
+		client:              client,
+		podService:          pods,
+		nodeService:         nodes,
+		pvService:           pvs,
+		pvcService:          pvcs,
+		storageClassService: storageClasss,
+		schedulerService:    schedulers,
+	}
+}
+
+// Get all resources from each service.
+func (s *Service) get(ctx context.Context) (*Resources, error) {
+	pods, err := s.podService.List(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to call list pods: %w", err)
+	}
+	nodes, err := s.nodeService.List(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to call list nodes: %w", err)
+	}
+	pvs, err := s.pvService.List(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to call list PersistentVolumes: %w", err)
+	}
+	pvcs, err := s.pvcService.List(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to call list PersistentVolumeClaims: %w", err)
+	}
+	scs, err := s.storageClassService.List(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to call list storageClasses")
+	}
+	ss := s.schedulerService.GetSchedulerConfig()
+
+	return &Resources{
+		PodList:          &pods.Items,
+		NodeList:         &nodes.Items,
+		PvList:           &pvs.Items,
+		PvcList:          &pvcs.Items,
+		StorageClassList: &scs.Items,
+		SchedulerConfig:  ss,
+	}, nil
+}
+
+func (s *Service) ExportAll(ctx context.Context) (*Resources, error) {
+	resources, err := s.get(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to export resources all: %w", err)
+	}
+	return resources, nil
+}
+
+// Import all resources from posted data.
+// (1) Restart scheduler based on the data.
+// (2) Apply each resource to the scheduler.
+//     * If UID is not nil, an error will occur. (try to find existing resource by UID)
+// (3) Get all resources. (Separated the get function to unify the struct format.)
+func (s *Service) ImportAll(ctx context.Context, resources *ResourcesApplyConfiguration) (*Resources, error) {
+
+	// TODO: Issue: #12 PR: #13
+	// if err := s.schedulerService.RestartScheduler(resources.SchedulerConfig); err != nil {
+	// 	klog.Warningf("failed to start scheduler with imported configuration: %v", err)
+	// 	return nil, xerrors.Errorf("restart scheduler with imported configuration: %w", err)
+	// }
+
+	for _, sc := range *resources.StorageClassList {
+		sc.ObjectMetaApplyConfiguration.UID = nil
+		_, err := s.storageClassService.Apply(ctx, &sc)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to apply StorageClass: %+v", err)
```

Comment:
- Errors should be wrapped with `%w`.
- In this project, error messages contain only "what did it do here". So you don't need `failed to` (This is not a rule in general, but at least in this project it is followed.)

```suggestion
			return nil, xerrors.Errorf("apply StorageClass: %w", err)
```